index.js

```diff
@@ -16,6 +16,16 @@
  */
 function Fireproof(firebaseRef, promise) {
 
+  if (!Fireproof.Promise) {
+
+    try {
+      Fireproof.Promise = Promise;
```

Comment:
I think just checking that it's a function would suffice.  Just a simple sanity check in case someone accidentally `Promise = true;`